src/uct/ib/ud/base/ud_iface.c

```diff
@@ -333,6 +333,10 @@ uct_ud_iface_set_event_cb(uct_ud_iface_t *iface, ucs_async_event_cb_t event_cb)
 
     ucs_assert(iface->async.event_cb != NULL);
 
+    if (!uct_ib_iface_device(&iface->super)->req_notify_cq_support) {
+        return UCS_ERR_UNSUPPORTED;
```

Comment:
maybe return UCS_OK so upper layer will not need to check for UCS_ERR_UNSUPPORTED?

Reply:
there is the arm_cq code that will be called if we return ok, moved this check outside this function, seems simpler, lmk